pkg/metadata/system.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2022 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package metadata
+
+import (
+	"syscall"
+
+	"github.com/parca-dev/parca-agent/pkg/buildinfo"
+	"github.com/prometheus/common/model"
+)
+
+var (
+	release, revision string
+)
+
+func int8SliceToString(arr []int8) string {
+	b := make([]byte, 0, len(arr))
+	for _, v := range arr {
+		// NUL byte, as it's a C string.
+		if v == 0 {
+			break
+		}
+		b = append(b, byte(v))
+	}
+	return string(b)
+}
+func kernelRelease() (string, error) {
+	var uname syscall.Utsname
+	err := syscall.Uname(&uname)
+	if err != nil {
+		return "", err
+	}
+
+	return int8SliceToString(uname.Release[:]), nil
+}
+
+// System provides metadata for the current system.
+func System() *Provider {
+	return &Provider{"system", func(_ int) (model.LabelSet, error) {
+		return model.LabelSet{
+			"kernel_release":     model.LabelValue(release),
+			"agent_vcs_revision": model.LabelValue(revision),
```

Comment:
I wonder if `_version` would be more expected, especially `agent_version`

Reply:
Perhaps `agent_revision`, so it's clear it's a VCS revision rather than semver?